### PR TITLE
feat: add agent options

### PR DESCRIPTION
**Pull request by Youkehai**
When your computer or server encounters network errors, timeouts, etc. while accessing Shopify, if you have a proxy server, you can set the proxy IP and port to resolve issues related to access timeouts and slowness.

### CODE REVIEW

**Comment by Youkehai** [comment]
Sorry, I modified my method

File: README.md [hunk 1]
```diff
@@ -78,6 +78,10 @@ Creates a new `Shopify` instance.
   attach to all outgoing requests, like `beforeRetry`, `afterResponse`, etc.
   Hooks should be provided in the same format that Got expects them and will
   receive the same arguments Got passes unchanged.
+- `agent` - Optional - A list of `got`
+  [request agent](https://github.com/sindresorhus/got/tree/v11.8.6?tab=readme-ov-file#agent)
+  This is necessary because a request to one protocol might redirect to another. 
+  In such a scenario, Got will switch over to the right protocol agent for you.
```
Comment (lpinca) [inline README.md:84]:
```suggestion
- `agent` - Optional - An object that is passed as the `agent` option to `got`.
  This allows to use a proxy server. See
  [Got documentation](https://github.com/sindresorhus/got/tree/v11.8.6?tab=readme-ov-file#proxies)
  for more details.
```

File: index.js [hunk 2]
```diff
@@ -9,6 +9,7 @@ const url = require('url');
 const pkg = require('./package');
 const resources = require('./resources');
 
+
```
Comment (lpinca) [inline index.js:12]:
```suggestion
```

File: index.js [hunk 3]
```diff
@@ -157,6 +158,10 @@ Shopify.prototype.request = function request(uri, method, key, data, headers) {
     method
```
Comment (lpinca) [inline index.js:158]:
```suggestion
    agent: this.options.agent,
    method
```

File: index.js [hunk 4]
```diff
@@ -157,6 +158,10 @@ Shopify.prototype.request = function request(uri, method, key, data, headers) {
...
   };
 
+  if (this.options.agent) {
+    options.agent = { ...this.options.agent };
+  }
+
```
Comment (lpinca) [inline index.js:164]:
```suggestion
```

File: index.js [hunk 5]
```diff
@@ -285,6 +290,10 @@ Shopify.prototype.graphql = function graphql(data, variables) {
     body: json ? this.options.stringifyJson({ query: data, variables }) : data
   };
 
+  if (this.options.agent) {
+    options.agent = { ...this.options.agent };
+  }
+
```
Comment (lpinca) [inline index.js:296]:
```suggestion
```

File: types/index.d.ts [hunk 6]
```diff
@@ -1,7 +1,7 @@
 // Type definitions for shopify-api-node
 // Project: shopify-api-node
 // Definitions by: Rich Buggy <rich@buggy.id.au>
-import { Hooks } from 'got';
+import { Hooks,Agents } from 'got';
```
Comment (lpinca) [inline types/index.d.ts:4]:
```suggestion
import { Hooks, Agents } from 'got';
```

File: types/index.d.ts [hunk 7]
```diff
@@ -789,6 +789,7 @@ declare namespace Shopify {
     shopName: string;
     timeout?: number;
     hooks?: Hooks;
+    agent?:Agents;
```
Comment (lpinca) [inline types/index.d.ts:792]:
```suggestion
    agent?: Agents;
```

File: types/index.d.ts [hunk 8]
```diff
@@ -801,6 +802,7 @@ declare namespace Shopify {
...
```
Comment (lpinca) [inline types/index.d.ts:805]:
```suggestion
    agent?: Agents;
```

File: types/index.test-d.ts [hunk 9]
```diff
@@ -27,6 +27,22 @@ new Shopify({
   }
 });
 
+// Can be constructed with Got proxy.
```
Comment (lpinca) [inline types/index.test-d.ts:30]:
```suggestion
// Accepts the `agent` option.
```

File: index.js [hunk 10]
```diff
@@ -285,6 +290,10 @@ Shopify.prototype.graphql = function graphql(data, variables) {
     body: json ? this.options.stringifyJson({ query: data, variables }) : data
```
Comment (lpinca) [inline index.js:290]:
```suggestion
    agent: this.options.agent
    body: json ? this.options.stringifyJson({ query: data, variables }) : data
```

**Review by lpinca — APPROVED** [review]

**Comment by lpinca** [comment]
Thank you.

**Pull request merged**